src/test/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesAppDeployerIntegrationTests.java

```diff
@@ -122,6 +123,120 @@ protected AppDeployer provideAppDeployer() {
 		return appDeployer;
 	}
 
+	@Test
+	public void testScaleStatefulApps() {
+		log.info("Testing {}...", "ScaleWithStatefulApp");
+		KubernetesDeployerProperties deployProperties = new KubernetesDeployerProperties();
+
+		ContainerFactory containerFactory = new DefaultContainerFactory(deployProperties);
+		KubernetesAppDeployer appDeployer = new KubernetesAppDeployer(deployProperties, kubernetesClient,
+				containerFactory);
+
+		AppDefinition definition = new AppDefinition(randomName(), null);
+		Resource resource = testApplication();
+
+		Map<String, String> props = new HashMap<>();
+		props.put(KubernetesAppDeployer.COUNT_PROPERTY_KEY, "3");
+		props.put(KubernetesAppDeployer.INDEXED_PROPERTY_KEY, "true");
+
+		AppDeploymentRequest request = new AppDeploymentRequest(definition, resource, props);
+
+		log.info("Deploying {}...", request.getDefinition().getName());
+		Timeout timeout = deploymentTimeout();
+		String deploymentId = appDeployer.deploy(request);
+		assertThat(deploymentId, eventually(hasStatusThat(
+				Matchers.hasProperty("state", is(deployed))), timeout.maxAttempts, timeout.pause));
+		assertThat(deploymentId, eventually(hasAppInstanceCount(is(3))));
+
+		// Ensure that a StatefulSet is deployed
+		Map<String, String> selector = Collections.singletonMap(SPRING_APP_KEY, deploymentId);
+		List<StatefulSet> statefulSets = kubernetesClient.apps().statefulSets().withLabels(selector).list().getItems();
+		assertNotNull(statefulSets);
+		assertTrue(statefulSets.size() == 1);
+		StatefulSet statefulSet = statefulSets.get(0);
+		StatefulSetSpec statefulSetSpec = statefulSet.getSpec();
+		Assertions.assertThat(statefulSetSpec.getPodManagementPolicy()).isEqualTo("Parallel");
+		Assertions.assertThat(statefulSetSpec.getReplicas()).isEqualTo(3);
+		Assertions.assertThat(statefulSetSpec.getServiceName()).isEqualTo(deploymentId);
+		Assertions.assertThat(statefulSet.getMetadata().getName()).isEqualTo(deploymentId);
+
+		log.info("Scale Down {}...", request.getDefinition().getName());
+		appDeployer.scale(deploymentId, 1);
+		assertThat(deploymentId, eventually(hasAppInstanceCount(is(1)), timeout.maxAttempts, timeout.pause));
+
+		statefulSets = kubernetesClient.apps().statefulSets().withLabels(selector).list().getItems();
+		assertTrue(statefulSets.size() == 1);
+		statefulSetSpec = statefulSets.get(0).getSpec();
+		Assertions.assertThat(statefulSetSpec.getReplicas()).isEqualTo(1);
+		Assertions.assertThat(statefulSetSpec.getServiceName()).isEqualTo(deploymentId);
+		Assertions.assertThat(statefulSet.getMetadata().getName()).isEqualTo(deploymentId);
+
+		appDeployer.undeploy(deploymentId);
+	}
+
+	@Test
+	public void testScaleWithStatelessApps() {
```

Comment:
consider `testScaleDeployment()` in the method signature and `ScaleDeployment` in the log message